### ISSUE TITLE
luarocks packaging requests

### ISSUE THREAD

**Issue by mrcjkb** [nvim-neorocks/rocks.nvim#111]
If you come across a plugin that isn't on luarocks.org, and you think it should be, you can leave a comment (or thumbs up of there's an existing request) here.

**Comment by glindstedt** [comment]
Migrated most of my config to use rocks.nvim and found these plugins to be missing from luarocks:

- [`nvim-neorg/neorg-telescope`](https://github.com/nvim-neorg/neorg-telescope)
- [`lukas-reineke/lsp-format`](https://github.com/lukas-reineke/lsp-format.nvim)
- [`lewis6991/gitsigns`](https://github.com/lewis6991/gitsigns.nvim)
- [`romgrk/barbar.nvim`](https://github.com/romgrk/barbar.nvim)
- [`tummetott/unimpaired.nvim`](https://github.com/Tummetott/unimpaired.nvim)
- [`nvimtools/none-ls.nvim`](https://github.com/nvimtools/none-ls.nvim)
- [`kevinhwang91/nvim-ufo`](https://github.com/kevinhwang91/nvim-ufo)
- [`nvim-neotest/neotest-go`](https://github.com/nvim-neotest/neotest-go)
- [`mg979/vim-visual-multi`](https://github.com/mg979/vim-visual-multi) (this is a vim-script plugin though, so not sure if applicable)

**Comment by mrcjkb** [comment]
https://github.com/nvim-neorg/neorg-telescope/pull/54

**Comment by datwaft** [comment]
Just found this issue and I found that [rktjmp/hotpot.nvim](https://github.com/rktjmp/hotpot.nvim/tree/master) is missing from luarocks.

**Comment by mrcjkb** [comment]
> Just found this issue and I found that [rktjmp/hotpot.nvim](https://github.com/rktjmp/hotpot.nvim/tree/master) is missing from luarocks.

I'm going to wait for https://github.com/nvim-neorocks/rocks.nvim/pull/236, because hotpot.nvim would need to be installed with `opt=true` and loaded manually, using `require("rocks").packadd(...)`.

**Comment by zackerydev** [comment]
https://github.com/nvim-neo-tree/neo-tree.nvim

Found this one today that is missing a luarocks package, last one I need for my config!

**Comment by mrcjkb** [comment]
https://github.com/nvim-neo-tree/neo-tree.nvim/pull/1434

**Comment by NTBBloodbath** [comment]
I would love to have this one to make some extensions for `rocks.nvim` like a `rocks-ui.nvim` module and easily write the UI/UX with it, `nui.nvim` is already on luarocks so it shouldn't take so long :p
- [nui-components.nvim](https://github.com/grapp-dev/nui-components.nvim)

**Comment by mrcjkb** [comment]
https://github.com/grapp-dev/nui-components.nvim/pull/22

**Comment by Anrock** [comment]
https://github.com/freddiehaddad/feline.nvim ?

**Comment by mrcjkb** [comment]
- https://github.com/freddiehaddad/feline.nvim/pull/71

**Comment by benlubas** [comment]
[molten-nvim](https://github.com/benlubas/molten-nvim)

_but_, it's a remote python plugin with a few python dependencies, so I haven't done it myself. I'm not sure if you have any advice on this? If there's going to be a lot of discussion we can move somewhere else to not pollute this thread.

**Comment by mrcjkb** [comment]
> [molten-nvim](https://github.com/benlubas/molten-nvim)
> 
> _but_, it's a remote python plugin with a few python dependencies, so I haven't done it myself. I'm not sure if you have any advice on this? If there's going to be a lot of discussion we can move somewhere else to not pollute this thread.

Looks like a nice challenge. I think it might need a luarocks python/venv build backend (for handling python dependencies), which doesn't seem to exist yet.

**Comment by benjstephenson** [comment]
[https://github.com/Olical/conjure](https://github.com/Olical/conjure) would be lovely :)

**Comment by calebeby** [comment]
These would be helpful:

https://github.com/nvim-telescope/telescope-fzf-native.nvim
https://github.com/nvim-telescope/telescope-fzy-native.nvim
https://github.com/natecraddock/telescope-zf-native.nvim

**Comment by mrcjkb** [comment]
- https://github.com/Olical/conjure/pull/579
- https://github.com/nvim-telescope/telescope-fzf-native.nvim/pull/127
- https://github.com/natecraddock/telescope-zf-native.nvim/pull/19

`telescope-fzy-native.nvim` uses submodules to depend on [fzy-lua-native](https://github.com/romgrk/fzy-lua-native).
`fzy-lua-native` is packaged in a very weird way, without any regard for potential module naming clashes (e.g. a `lua/native.lua` file, wtf?).
I don't think it makes sense to attempt to package that. A better alternative might be to rewrite `telescope-fzy-native.nvim`, so that it depends on [fzy](https://luarocks.org/modules/sethwarn/fzy), which is packaged properly, instead.

**Comment by boltlessengineer** [comment]
`treesj` is already on luarocks, but its dependency should be updated. `nvim-treesitter` should be removed from dependency[^1].

[^1]: https://github.com/Wansmer/treesj/issues/153#issuecomment-2132078689

**Comment by mrcjkb** [comment]
@boltlessengineer Updated :smile: 

@craigmac sorry for taking so long to respond. I hope to get back to this soon :sweat_smile:

**Comment by Freed-Wu** [comment]
https://github.com/black-desk/fcitx5-ui.nvim A neovim plugin to input Chinese characters

Related issue: https://github.com/black-desk/fcitx5-ui.nvim/issues/1

**Comment by simifalaye** [comment]
Thanks a lot for all the work on this. I'd like to request these:
- https://github.com/folke/ts-comments.nvim
- https://github.com/rafamadriz/friendly-snippets
- https://github.com/garymjr/nvim-snippets
- https://github.com/MeanderingProgrammer/markdown.nvim

Also, I saw you've been working on getting cmp to update its releases so hopefully the maintainer will notice soon. Thanks!

**Comment by mrcjkb** [comment]
> One thing on the friendly-snippets package: I'm not seeing the "snippets" directory which contains all the snippets it provides on my machine after installing the plugin.

Ah, that's not a standard neovim directory. I'll have to look into adding the ability to add additional directories to the NURR CI

**Comment by mrcjkb** [comment]
@simifalaye `friendly-snippets` should be fixed now.

**Comment by simifalaye** [comment]
@mrcjkb Any news on [markdown.nvim](https://github.com/MeanderingProgrammer/markdown.nvim)?

Also, I saw that the gitsigns PR was merged but it's just waiting for a new release to be made?

Thanks very much for all your work, you are a machine!

**Comment by benlubas** [comment]
Could you possibly help out with [benlubas/neorg-se](https://github.com/benlubas/neorg-se)?

It's a plugin that bundles a rust binary using mlua and the `luarocks-build-rust-mlua` backend. I've been unsuccessful so far in getting lazy to install and build it. I can install with `luarocks --local --lua-version 5.1 install neorg-se` without errors. But even that doesn't seem like it's building the rust module, and I don't see the `libneorg_se.so` file that I expect to see anywhere in `~/.luarocks` or `~/.local/share/nvim/lazy(-rocks)`

**Comment by mrcjkb** [comment]
I'm away from my laptop this week.

> Any news on markdown.nvim?

I'm planning on publishing the tree-sitter parsers with 0-ver versions (in anticipation nvim-treesitter going semver one day).
When that is done, markdown.nvim can add them as a dependency. 
If it depended on the current `dev` packages, it wouldn't be possible to install a non-dev version of markdown.nvim.

**Comment by mrcjkb** [comment]
> Could you possibly help out with benlubas/neorg-se?

I can see if it works with rocks.nvim when I'm back on my laptop.

It looks like whatever lazy.nvim is using to install it can't find the `luarocks-build-rust-mlua` build backend, which could have something to do with how lazy.nvim installs dependencies and rocks into separate trees (iirc; not sure about this).

You could try working around this by installing `luarocks-build-rust-mlua` locally:
`luarocks --local install luarocks-build-rust-mlua`

**Comment by benlubas** [comment]
Thank you, that did allow me to install it! I can get another rocks.nvim user to test it for me so don't worry about that. thank you for the suggestion!

**Comment by simifalaye** [comment]
This would be great: https://github.com/creativenull/efmls-configs-nvim

**Comment by vimproved** [comment]
would love to see https://github.com/udayvir-singh/tangerine.nvim and https://github.com/udayvir-singh/hibiscus.nvim :)

**Comment by mrcjkb** [comment]
@simifalaye https://github.com/nvim-neorocks/nurr/pull/20

@vimproved I'll look into opening some PRs. By the looks of it, bootstrapping would be unnecessary with rocks.nvim.

**Comment by Freed-Wu** [comment]
> firenvim is going to be a challenge, because it uses an autoload function to install.

Do you mean `call firenvim#install(0)`? Every time we install a new browser, it should be execute to let it find it. Maybe we don't need to ru this function during installing, and let user do it when they install a browser.

> nvim-treesitter going semver one day

nvim-treesitter disappear from luarocks. Anyone know what happened?

**Comment by mrcjkb** [comment]
> Do you mean call firenvim#install(0)? Every time we install a new browser, it should be execute to let it find it. Maybe we don't need to ru this function during installing, and let user do it when they install a browser.

That sounds reasonable to me. 
Perhaps that info should be added to the firenvim readme (if not already present).
Still, it would be nice to have it auto install for existing browsers (but not a must-have). But I suppose since this doesn't get cleaned up when uninstalling the plugin, it might be better to have users call it explicitly. 

> nvim-treesitter disappear from luarocks. Anyone know what happened?

Yep, we removed it because it's very outdated and causes a lot of problems and conflicts with the luarocks tree-sitter parsers. There's a `nvim-treesitter-legacy-api` which is basically nvim-treesitter without `TSInstall` and without queries, which you can use for plugins that depend on the deprecated nvim-treesitter API.
If you want nvim-treesitter with `TSInstall`, you need to use rock-git.nvim. But keep in mind, the `master` branch may cause conflicts and the `main` branch requires nvim nightly.

**Comment by Freed-Wu** [comment]
> added to the firenvim readme

Add rocks.nvim to https://github.com/glacambre/firenvim#installing :smile:

> very outdated

Perhaps a scm is possible?

**Comment by mrcjkb** [comment]
> Perhaps a scm is possible?

I guess am scm version for the `main` branch would be reasonable.
`TSInstall` would still be broken though, because it makes assumptions about where the queries are installed. So you might as well use the `scm` version of the nvim-treesitter-legacy-api.

**Comment by mrcjkb** [comment]
@Freed-Wu [here's an experiment](https://luarocks.org/modules/neorocks/nvim-treesitter)...

**Comment by Freed-Wu** [comment]
So `TSInstall XXX` is `Rocks install tree-sitter-XXX`, right? `TSInstall all` is search all `tree-sitter-XXX` in luarocks.org and install them, right?

**Comment by mrcjkb** [comment]
> So TSInstall XXX is Rocks install tree-sitter-XXX, right? TSInstall all is search all tree-sitter-XXX in luarocks.org and install them, right?

Yes - If rocks.nvim is installed.

**Comment by Freed-Wu** [comment]
I see. Original `TSInstall` is in parallel and now its behaviour is like `TSInstallSync`.

**Comment by bottd** [comment]
Would be nice to get nushell treesitter on luarocks
https://github.com/nushell/tree-sitter-nu

**Comment by mrcjkb** [comment]
Opened an issue with tree-sitter-nu for now

- https://github.com/nushell/tree-sitter-nu/issues/103

Added quicker.nvim to the NURR.

**Comment by Nsidorenco** [comment]
Would be nice to have the following :)
https://github.com/Issafalcon/neotest-dotnet
https://github.com/voldikss/vim-floaterm

**Comment by mrcjkb** [comment]
@Nsidorenco 

- https://github.com/Issafalcon/neotest-dotnet/pull/115
- [ ] Will look at vim-floatterm later (I'm travelling right now)

**Comment by simifalaye** [comment]
@mrcjkb Wanted to bring your attention to this: [blink.cmp](https://github.com/Saghen/blink.cmp)

From the issues, it looks like the maintainer is open to publishing with luarocks but wanted to ping you in case you may have any guidance to give. Thanks!

**Comment by mrcjkb** [comment]
@simifalaye thanks for the heads-up. I'll keep my eye on the issue 😃

**Comment by Freed-Wu** [comment]
https://github.com/wasm-lsp/tree-sitter-wasm

Tree-sitter for wasm :smile:

https://github.com/icedman/nvim-textmate/issues/8

A neovim plugin to use textmate theme and grammar

**Comment by mrcjkb** [comment]
@Freed-Wu any idea why tree-sitter-wasm isn't supported by nvim-treesitter?
I'll be busy this week, but I can likely look into those sometime after Sunday.
